polaris-core/src/main/java/org/apache/polaris/core/storage/aws/AwsStorageConfigurationInfo.java

```diff
@@ -49,23 +49,30 @@ public class AwsStorageConfigurationInfo extends PolarisStorageConfigurationInfo
   @JsonProperty(value = "userARN")
   private @Nullable String userARN = null;
 
+  /** User ARN for the service principal */
+  @JsonProperty(value = "region")
+  private @Nullable String region = null;
+
   @JsonCreator
   public AwsStorageConfigurationInfo(
       @JsonProperty(value = "storageType", required = true) @NotNull StorageType storageType,
       @JsonProperty(value = "allowedLocations", required = true) @NotNull
           List<String> allowedLocations,
-      @JsonProperty(value = "roleARN", required = true) @NotNull String roleARN) {
-    this(storageType, allowedLocations, roleARN, null);
+      @JsonProperty(value = "roleARN", required = true) @NotNull String roleARN,
+      @JsonProperty(value = "region", required = false) @NotNull String region) {
```

Comment:
Why `@NotNull` here while the property is nullable?

Reply:
Good catch, I thought it had to be `NotNull` for the `JsonCreator` but in fact that's not true. Fixed.